batchflow/models/validator/validator.py

```diff
@@ -0,0 +1,337 @@
+        for cls in inspect.getmro(meth.__self__.__class__):
+            if cls.__dict__.get(meth.__name__) is meth:
+                return cls
+        meth = meth.__func__  # fallback to __qualname__ parsing
+    if inspect.isfunction(meth):
+        cls = getattr(inspect.getmodule(meth),
+                      meth.__qualname__.split('.<locals>', 1)[0].rsplit('.', 1)[0])
+        if isinstance(cls, type):
+            return cls
+    return getattr(meth, '__objclass__', None)  # handle special descriptor objects
+
+class Validator:
+    """ Validator: interface for loaders, train, inference and metrics evaluation.
+
+    **Attributes**
+
+    config_path : str
+        path to yaml file with config of the following structure:
+        ```
+        train: (optional)
+            dataset:
+                - <dataset_param_0>: <value_0>
+                ...
+        pretrained: (optional)
+            path: <model_path>
+        test: (optional)
+            dataset:
+                - <dataset_param_0>: <value_0>
+                ...
+            metrics: (metics from batchflow)
+                - <classification|segmentation|mask|instance|regression>
+                    <classification_kwarg_0>: <value_0>
+                    <classification_kwarg_1>: <value_1>
+                    ...
+                    evaluate:
+                        - <metric_name_0>:
+                            <metric_kwarg_0>: <value_0>
+                            <metric_kwarg_1>: <value_1>
+                        - <metric_name_1>:
+                            ...
+                        ...
+            custom_metrics: (optional, metrics defined in ModelAPI child-class)
+                - <custom_metric_0>
+                - <custom_metric_1>
+                ...
+        ```
+    name : str
+        name of the current model (from yaml)
+    task : str
+        name of the current task (from yaml)
+    """
+    def __init__(self, config=None, **kwargs):
+        if isinstance(config, str):
+            self.config_path = config
+            with open(config) as file:
+                self.config = yaml.load(file, Loader=yaml.Loader)
+        elif isinstance(config, dict):
+            self.config_path = None
+            self.config = deepcopy(config)
+        elif isinstance(config, Config):
+            self.config_path = None
+            self.config = deepcopy(config.config)
+        elif config is None:
+            self.config = {}
+
+        self.config = {**self.config, **kwargs}
+
+        if 'train' in self.config and 'pretrained' in self.config:
+            raise ValueError("Both 'train' and 'pretrained' was founded.")
+
+        self._pretrained = self.config.get('pretrained', {})
+        if 'pretrained' in self.config:
+            if self._pretrained is None:
+                self._pretrained = {}
+            elif isinstance(self._pretrained, str):
+                self._pretrained = {'path': self._pretrained}
```

Comment:
Looks cumbersome, maybe:
`self._pretrained = {'path': config['pretrained']} if isinstance(config.get('pretrained'), str) else config.get('pretrained', {})`
split into multiple lines. Same for `self._train_ds`, `self._test_ds`

Reply:
Done